src/App.jsx

```diff
@@ -14,45 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_NAME = 'name';
+const SORT_LENGTH = 'length';
+
+function getPreparedGoods(goods, { sortField, isReversed }) {
+  const preparedGoods = [...goods];
+
+  if (sortField === SORT_NAME) {
+    preparedGoods.sort((prod1, prod2) => prod1.localeCompare(prod2));
+  }
+
+  if (sortField === SORT_LENGTH) {
+    preparedGoods.sort(
+      (product1, product2) => product1.length - product2.length,
+    );
+  }
+
+  if (isReversed) {
+    return preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    isReversed,
+  });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const isChanged = sortField !== '' || isReversed;
+
+  const onReset = () => {
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn(
+            { 'is-light': sortField !== SORT_NAME },
+            'button is-info',
+          )}
+          onClick={() => {
+            setSortField(SORT_NAME);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            { 'is-light': sortField !== SORT_LENGTH },
+            'button is-success',
+          )}
+          onClick={() => {
+            setSortField(SORT_LENGTH);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn(
+            { 'is-light': isReversed === false },
+            'button is-warning',
+          )}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
```

Comment:
```suggestion
          onClick={() => setIsReversed(prevIsReversed => !prevIsReversed)}
```
better to use prev state in this case.